File: back-end/src/main/java/server/model/Zvanje.java
package server.model;

import java.util.Date;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Zvanje {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	@Column(nullable=true)
	private Date datumIzbora;
	@Column(nullable=true)
	private Date datumPrestanka;
	
	@ManyToOne(optional = false)
	private TipZvanja tipZvanja;
	
	@ManyToOne(optional = false)
	private NaucnaOblast naucnaOblast;
	
	@ManyToOne(optional=false)
	private Nastavnik nastavnik;
	
    @Column(nullable = false)
    private Boolean vidljiv = true;
}

File: front-end/src/app/models/Zvanje.ts
import { Nastavnik } from "./Nastavnik";
import { NaucnaOblast } from "./NaucnaOblast";
import { TipZvanja } from "./TipZvanja";


export interface Zvanje {
  id: number;
  datumIzbora: Date;
  datumPrestanka: Date;
  tipZvanja: TipZvanja;
  naucnaOblast: NaucnaOblast;
  nastavnik: Nastavnik;
  vidljiv: boolean;
}
